flocker/node/agents/blockdevice.py

```diff
@@ -1395,6 +1418,61 @@ def _manifestation_from_volume(volume):
     return Manifestation(dataset=dataset, primary=True)
 
 
+def _ext4_fs_uuid(device_path):
+    """
+    Get the EXT4 filesystem UUID from the filesystem on the device at the given
+    path.
+
+    :param FilePath device_path: The path to the device on which the filesystem
+        resides.
+
+    :return: A ``UUID`` instance giving the UUID from the filesystem.
+    :rtype: ``UUID``
+    """
+    blkid = _blkid(
```

Comment:
https://www.kernel.org/pub/linux/utils/util-linux/v2.21/libblkid-docs/libblkid-Cache.html suggest the results will always be up-to-date, if we have read access to the device (which we will, running as root).